net/FlatBuffers/ByteBuffer.cs

```diff
@@ -886,6 +891,60 @@ public int Put<T>(int offset, Span<T> x)
             }
             return offset;
         }
+#else
+        /// <summary>
+        /// Copies an array segment of type T into this buffer, ending at the given
+        /// offset into this buffer. The starting offset is calculated based on the length
+        /// of the array segment and is the value returned.
+        /// </summary>
+        /// <typeparam name="T">The type of the input data (must be a struct)</typeparam>
+        /// <param name="offset">The offset into this buffer where the copy will end</param>
+        /// <param name="x">The array to copy data from</param>
+        /// <returns>The 'start' location of this buffer now, after the copy completed</returns>
+        public int Put<T>(int offset, ArraySegment<T> x)
```

Comment:
a fair bit of code in here is shared with the span case, does it make sense to lift some of it out into common functions, assuming it won't affect performance too much?